### DISCUSSION TITLE
Using gitoxide crates

### DISCUSSION THREAD

**Discussion by darleybarreto** [gitbutlerapp/gitbutler#2565] [Ideas]
Hey there, 

Have you consider using [gitoxide](https://github.com/Byron/gitoxide) crates as experimental features?
They are probably incomplete in functionalities needed by gitbutler, but I imagine that Sebastian (main contributor) might be open to add a tracklist for the missing pieces. He usually does something like "Oxidize XXX".

**Comment by schacon** [comment]
We have looked at this, I think because it can be threadsafe where libgit2 cannot. Nikita setup a sort of virtual git call layer so we could use gitoxide for some git operations and libgit2 or fork/exec with others, depending on what was better or faster. So yes, we are interested in doing that at some point, or at least for some commands or operations, we just haven't gotten there yet.

**Comment by Qix-** [comment]
We're actually setting things up so that other Git backends can be added - see the `gitbutler-git` crate, which we're moving to. It's an abstraction over git2 and a fork/exec out to the `git` cli, and gitoxide could be a supported backend in the future.

Will take some time to flesh it out but should be possible :)

**Comment by darleybarreto** [comment]
Hey folks,

I'm off due to Carnival and just found out that Sebastian actually did the discovery in [this](https://github.com/Byron/gitoxide/issues/1287) issue.

**Comment by Byron** [comment]
I think it's fair to close this discussion as we are in the last stretch of replacing `git2` entirely with `gix`.
